CMakeLists.txt

```diff
@@ -1,19 +1,52 @@
-cmake_minimum_required (VERSION 2.8.11)
+cmake_minimum_required (VERSION 3.5)
+
+if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake")
+endif()
+
 project (signalrclient)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -fPIC -L -lcpprest")
+if(NOT WIN32)
+  set(EXTRA_FLAGS "-std=c++11 -fPIC -DNO_SIGNALRCLIENT_EXPORTS")
+else()
+  set(EXTRA_FLAGS "-DSIGNALRCLIENT_EXPORTS")
+endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${EXTRA_FLAGS}")
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${CMAKE_LDFLAGS}")
+
+find_package(cpprestsdk CONFIG REQUIRED)
+set(CPPREST_LIB "cpprestsdk::cpprest")
 
-set(CPPREST_INCLUDE_DIR "" CACHE FILEPATH "Path to casablanca include dir")
+include_directories (include)
 
-include_directories (
-include
-"${CPPREST_INCLUDE_DIR}")
+# TODO: We shouldn't use this, it makes the dll/lib export all symbols
+# We need this for testing, but we might just add all the files to the test project manually
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
```

Comment:
We recommend against using `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` because it has a lot of drawbacks and problems.

We would instead recommend one of:
1. Apply dllexport as needed (remember that you can apply it to entire classes)
2. Provide a .def file listing all the names to export. This file can simply be added to the `add_library()` call and CMake will take care of passing it to the linker.
3. Only support static libraries on Windows (though this has obvious problems if you're expecting to P/Invoke from .NET)